### ISSUE TITLE
data.json does not sync between instances

### ISSUE THREAD

**Issue by anoadragon453** [HelloZeroNet/ZeroNet#1383]
### Step 1: Please describe your environment

  * ZeroNet version: 0.6.2 (rev 3408 on both computers)
  * Operating system: Linux
  * Web browser: Chromium
  * Tor status: available
  * Opened port: yes (on remote computer)

### Step 2: Describe the problem:

Computer 1:

* Contains site.
* VPN
* Port closed
* Tor available

Computer 2 (remote):

* Wants site
* Ports 10000-40000 tcp/udp open
* Tor available

If I create a new site from ZeroHello on Computer 1 and visit that site on Computer 2, all is well, it loads nearly instantly.

However, I'm trying to publish a merger site to the remote computer. content.json and index.html send over fine, but publishing the data/user/xxx/data.json file does not work. On Computer 1, the following appears in the log:

```
[15:01:48] Site:15GWMQ..mRnr Publishing data/users/xxx/content.json to 5/4 peers (connected: 1) diffs: [u'data.json'] (9.30k)...
[15:02:03] Site:15GWMQ..mRnr [FAILED] x.x.x.x:28016: None
[15:02:16] Site:15GWMQ..mRnr [FAILED] x.x.x.x:16990: {'exception': 'Timeout'}  # Not sure why it's publishing to two ports on the same IP here
[15:02:46] Site:15GWMQ..mRnr [FAILED] bhv7r6tgzxm2byqd.onion:29593: None
[15:02:46] Site:15GWMQ..mRnr [FAILED] dnskvt25zkla3eur.onion:15441: None
[15:02:46] Site:15GWMQ..mRnr Successfuly data/users/xxx/content.json published to 0 peers, publishing to 5 more peers in the background
```

The contents of the user's `content.json`:

```
{
 "address": "yyy",
 "cert_auth_type": "web",
 "cert_sign": "Gx/IDfK+18c0cCde/ghhCu+G7YOc+8Q4zmF0Q+BU6iaMEoJnUpdEhrr3g1Kl4UQnghsBuW4I3OxAGA1FnLkeoE0=",
 "cert_user_id": "xyx@zeroid.bit",
 "files": {
  ...some files here...
  "data.json": {
   "sha512": "a9a1914200236e68d6c4afecd7fa55134c1c30ede7c0491724b35fcb7e247acd",
   "size": 8177
  }
 },
 "files_optional": {
  ...some files here...
 },
 "inner_path": "data/users/xxx/content.json",
 "modified": 1523386907,
 "signs": {
  "xxx": "HKANPQh9ABQzUmLflk7uSs4mN6xWFBIan4vpoZr5bqW7CR1Kc2IZ2nJJ3y0tV6B6N3dwAcapJ2SgK2VlAVxWP8U="
 }
}
```

`data.json` is valid JSON.

The `content.json` of the site is as follows:

```
{
 "address": "yyy",
 "address_index": 56380503,
 "background-color": "#FFF",
 "clone_root": "",
 "cloneable": true,
 "description": "",
 "favicon": "img/favicon.svg",
 "files": {
  "data-default/users/content.json-default": {
   "sha512": "8a6052cc42b4bf351cc70eb3bf154b74a8eb693382995262eb0edb9ee0c498fc",
   "size": 582
  },
  "img/favicon.svg": {
   "sha512": "2ae0beba523a3e4c156f789173eec9065bb903cc2f73ec4e5bbcb130e050078e",
   "size": 2221
  },
  "index.html": {
   "sha512": "642c967389e3e4870ea0052f3228f9d9a553375d938004dbeb89d9fa0daa9805",
   "size": 494
  },
  "js/ZeroFrame.js": {
   "sha512": "d64a15624523379ee413c3a4a63cd6eba7934bfb85ce35c95199874fb02a661c",
   "size": 4000
  }
 },
 "ignore": "data/.*",
 "includes": {
  "data/users/content.json": {
   "signers": [],
   "signers_required": 1
  }
 },
 "inner_path": "content.json",
 "installed": false,
 "merged_type": "zzz",
 "modified": 1523333129,
 "postmessage_nonce_security": true,
 "signers_sign": "HOqqx698s1GXMSBRGYEdS+/xSHrXNJoUS+GmoyTjOmDsHTtK/3bx6fKGgqNz4YOEJz7daRZHtFEgo+0MPOlimhw=",
 "signs": {
  "15GWMQgWrxJvm41ak7sWAaSRXw1ThMmRnr": "G0ND/KY/HswHa+molrTWUp8qaxPmgE+Qtx/mBEfxCPEnEJ22+X+hp5+DS3q/XgOJaClF3lWrUwZyp2sNovYzYkk="
 },
 "signs_required": 1,
 "title": "Merger",
 "translate": ["js/all.js"],
 "zeronet_version": "0.6.2"
}
```

It doesn't publish over Tor (tried Tor always on both servers, they see each other but keep getting `None` result) nor over clearnet. Since things work with other sites it makes me think it's something to do with either the site `content.json` or the user's. In the site sidebar on the remote server it says it needs `data/users/xxx/content.json` but it never gets it. I've tried the same setup with a couple other computers and it's the same story, none of them get the `data.json` file.

Any ideas? Thanks!

**Comment by anoadragon453** [comment]
Another question I have that may be related: If a merger site has the default 10MB size limit, and a user is trying to publish content that is, say 15MB. Is there any automatic fashion for the merger site's size limit to be increased while the user is publishing, or do we have to manually `fileGet` the merger's user `content.json`, modifying the `max_size`, signing and then sign/publish the content?

**Comment by anoadragon453** [comment]
Ok, at this point I've found that all user data syncs over to the remote computer *except* `data.json`, which is what the site actually uses to display information. The size of `data.json` (6KiB) is relatively small compared to user data, (~2MB). Not sure why it wouldn't sync otherwise...

**Comment by anoadragon453** [comment]
`--debug` reveals some more information. This is what I get on the remote computer (Computer 2) after publishing on the local computer (Computer 1):

```
[10:57:31] WorkerManager:1146pu..rMaE New task: data/users/xxx/data.json, peer lock: None, priority: 3, optional_hash_id: None, tasks started: 1
[10:57:31] WorkerManager:1146pu..rMaE Starting workers, tasks: 1, peers: 0, workers: 0
[10:57:31] WorkerManager:1146pu..rMaE Added worker: 73.66.126.33:15441, workers: 1/5
[10:57:31] WorkerManager:1146pu..rMaE Added worker: 3krzvwaaz2wusilh.onion:29593, workers: 2/5
[10:57:31] WorkerManager:1146pu..rMaE Added worker: 4y4v2bidyockez72.onion:29593, workers: 3/5
[10:57:31] WorkerManager:1146pu..rMaE Added worker: 162.158.89.247:28016, workers: 4/5
[10:57:31] WorkerManager:1146pu..rMaE Added worker: bhv7r6tgzmm2byqd.onion:29593, workers: 5/5
[10:57:31] WorkerManager:1146pu..rMaE 73.66.126.33:15441: Verify failed: data/users/xxx/data.json, error: Download failed, failed peers: 0, workers: 0
[10:57:32] WorkerManager:1146pu..rMaE 3krzvwaaz2wusilh.onion:29593: Verify failed: data/users/xxx/data.json, error: Download failed, failed peers:1, workers: 3
[10:57:33] WorkerManager:1146pu..rMaE 73.66.126.33:15441: No task found, stopping
[10:57:33] WorkerManager:1146pu..rMaE Removed worker, workers: 4/5
[10:57:33] WorkerManager:1146pu..rMaE 162.158.89.247:28016: Verify failed: data/users/xxx/data.json, error: Download failed, failed peers: 2, workers: 2
[10:57:33] WorkerManager:1146pu..rMaE 3krzvwaaz2wusilh.onion:29593: No task found, stopping
[10:57:33] WorkerManager:1146pu..rMaE Removed worker, workers: 3/5
[10:57:34] WorkerManager:1146pu..rMaE 162.158.89.247:28016: No task found, stopping
[10:57:34] WorkerManager:1146pu..rMaE Removed worker, workers: 2/5
[10:58:00] WorkerManager:1146pu..rMaE Slow task: data/users/xxx/data.json, (workers: 2, optional_hash_id: None, peers: 0, failed: 3, asked: 0)
[10:58:00] Site:1146pu..rMaE Tracker result: zero://zero.booth.moe#f36ca555bee6ba216b14d11f38c16f7769ff064e0e37d887603548cc2e64191d:443 (sites: 1, new: 0)
[10:58:00] WorkerManager:1146pu..rMaE Starting workers, tasks: 1, peers: 0, workers: 2
[10:58:00] WorkerManager:1146pu..rMaE Added worker: 3krzvwaaz2wusilh.onion:29593, workers: 3/5
[10:58:00] WorkerManager:1146pu..rMaE Added worker: nsizc3h4pwq6xdkq.onion:29593, workers: 4/5
[10:58:00] WorkerManager:1146pu..rMaE Added worker: 162.158.89.247:28016, workers: 5/5
[10:58:00] WorkerManager:1146pu..rMaE Added worker: ndne6fyg5ecd3p2a.onion:29593, workers: 6/5
[10:58:01] WorkerManager:1146pu..rMaE 3krzvwaaz2wusilh.onion:29593: No task found, stopping
[10:58:01] WorkerManager:1146pu..rMaE Removed worker, workers: 5/5
[10:58:01] WorkerManager:1146pu..rMaE 162.158.89.247:28016: No task found, stopping
[10:58:01] WorkerManager:1146pu..rMaE Removed worker, workers: 4/5
[10:58:03] Site:1146pu..rMaE Small number of peers detected...query all of peers using pex    
[10:58:05] Site:1146pu..rMaE Connected before: 5, after: 6. Check site: False.           
[10:58:15] WorkerManager:1146pu..rMaE Slow task: data/users/xxx/data.json, (workers: 4, optional_hash_id: None, peers: 0, failed: 3, asked: 0)     
[10:58:15] WorkerManager:1146pu..rMaE Starting workers, tasks: 1, peers: 0, workers: 4   
[10:58:15] WorkerManager:1146pu..rMaE Added worker: 73.66.126.33:15441, workers: 5/5     
[10:58:15] WorkerManager:1146pu..rMaE Added worker: m5eynz2x3onhxvtq.onion:29593, worker$: 6/5                             
[10:58:16] WorkerManager:1146pu..rMaE 73.66.126.33:15441: No task found, stopping        
[10:58:16] WorkerManager:1146pu..rMaE Removed worker, workers: 5/5                       
[10:58:20] WorkerManager:1146pu..rMaE ndne6fyg5ecd3p2a.onion:29593: Verify failed: data/users/xxx/data.json, error: Download failed, failed peers: 3, workers: 4
[10:58:21] WorkerManager:1146pu..rMaE ndne6fyg5ecd3p2a.onion:29593: No task found, stopping                                                           
[10:58:21] WorkerManager:1146pu..rMaE Removed worker, workers: 4/5                       
[10:58:30] WorkerManager:1146pu..rMaE Slow task: data/users/xxx$NvENj/data.json, (workers: 4, optional_hash_id: None, peers: 0, failed: 4, asked: 0)     
[10:58:30] WorkerManager:1146pu..rMaE Starting workers, tasks: 1, peers: 0, workers: 4   
[10:58:30] WorkerManager:1146pu..rMaE Added worker: 73.66.126.33:15441, workers: 5/5     
[10:58:30] WorkerManager:1146pu..rMaE Added worker: 3krzvwaaz2wusilh.onion:29593, worker$: 6/5                                                                                    
[10:58:31] WorkerManager:1146pu..rMaE 73.66.126.33:15441: No task found, stopping        
[10:58:31] WorkerManager:1146pu..rMaE Removed worker, workers: 5/5                       
[10:58:31] WorkerManager:1146pu..rMaE 3krzvwaaz2wusilh.onion:29593: No task found, stopping                                               
[10:58:31] WorkerManager:1146pu..rMaE Removed worker, workers: 4/5                       
[10:58:45] WorkerManager:1146pu..rMaE m5eynz2x3onhxvtq.onion:29593: Verify failed: data/$sers/xxx/data.json, error: Download failed, failed peers: 4, workers: 3  
[10:58:45] WorkerManager:1146pu..rMaE Timeout, Skipping: {'optional_hash_id': None, 'sit$': <Site 1146pu..rMaE>, 'done': False, 'size': 1058, 'inner_path': u'data/users/xxx/data.json', 'peers': None, 'time_started': 1523609851.607931, $time_action': None, 'priority': 3, 'failed': [<Peer:73.66.126.33>, <Peer:3krzvwaaz2wusil$.onion>, <Peer:162.158.89.247>, <Peer:ndne6fyg5ecd3p2a.onion>, <Peer:m5eynz2x3onhxvtq.on$on>], 'workers_num': 3, 'time_added': 1523609851.606069, 'evt': <gevent.event.AsyncResul$ object at 0x7f408abd11d0>}                                                              
[10:58:45] WorkerManager:1146pu..rMaE bhv7r6tgzmm2byqd.onion:29593: Force skipping       
[10:58:46] WorkerManager:1146pu..rMaE m5eynz2x3onhxvtq.onion:29593: No task found, stopping                                     
[10:58:46] WorkerManager:1146pu..rMaE Removed worker, workers: 3/5                       
[10:58:58] WorkerManager:1146pu..rMaE Starting workers, tasks: 1, peers: 0, workers: 3   
[10:58:58] WorkerManager:1146pu..rMaE Added worker: 73.66.126.33:15441, workers: 4/5
[10:58:58] WorkerManager:1146pu..rMaE Added worker: 3krzvwaaz2wusilh.onion:29593, worker
[10:58:58] WorkerManager:1146pu..rMaE Added worker: mep3zrh575k5fw6g.onion:29593, workers: 6/5                  
[10:58:58] WorkerManager:1146pu..rMaE 73.66.126.33:15441: No task found, stopping        
[10:58:58] WorkerManager:1146pu..rMaE Removed worker, workers: 5/5
[10:58:58] WorkerManager:1146pu..rMaE 3krzvwaaz2wusilh.onion:29593: No task found, stopping                                
[10:58:58] WorkerManager:1146pu..rMaE Removed worker, workers: 4/5                      
[10:59:00] WorkerManager:1146pu..rMaE bhv7r6tgzmm2byqd.onion:29593: Verify failed: data/users/xxx/data.json, error: Download failed, failed peers: 5, workers: 3
[10:59:00] WorkerManager:1146pu..rMaE Starting workers, tasks: 1, peers: 0, workers: 4
[10:59:00] WorkerManager:1146pu..rMaE Added worker: 73.66.126.33:15441, workers: 5/5
[10:59:00] WorkerManager:1146pu..rMaE Added worker: 3krzvwaaz2wusilh.onion:29593, workers: 6/5
[10:59:00] WorkerManager:1146pu..rMaE 73.66.126.33:15441: No task found, stopping
[10:59:00] WorkerManager:1146pu..rMaE Removed worker, workers: 5/5
[10:59:00] WorkerManager:1146pu..rMaE 3krzvwaaz2wusilh.onion:29593: No task found, stopping
[10:59:00] WorkerManager:1146pu..rMaE Removed worker, workers: 4/5
[10:59:01] WorkerManager:1146pu..rMaE bhv7r6tgzmm2byqd.onion:29593: No task found, stopping
[10:59:01] WorkerManager:1146pu..rMaE Removed worker, workers: 3/5
[10:59:01] WorkerManager:1146pu..rMaE nsizc3h4pwq6xdkq.onion:29593: Force skipping
[10:59:01] WorkerManager:1146pu..rMaE bhv7r6tgzmm2byqd.onion:29593: No task found, stopping
[11:02:01] WorkerManager:1146pu..rMaE Removed worker, workers: 1/5                       
[11:02:01] WorkerManager:1146pu..rMaE 162.158.89.247:28016: No task found, stopping
[11:02:01] WorkerManager:1146pu..rMaE Removed worker, workers: 0/5
[11:02:05] WorkerManager:1146pu..rMaE Timeout, Skipping: {'optional_hash_id': None, 'site': <Site 1146pu..rMaE>, 'done': False, 'size': 1058, 'inner_path': u'data/users/xxx/data.json', 'peers': None, 'time_started': 1523609851.607931, 'time_action': None, 'priority': 3, 'failed': [<Peer:73.66.126.33>, <Peer:3krzvwaaz2wusilh.onion>, <Peer:162.158.89.247>, <Peer:ndne6fyg5ecd3p2a.onion>, <Peer:m5eynz2x3onhxvtq.onion>, <Peer:bhv7r6tgzmm2byqd.onion>, <Peer:4y4v2bidyockez72.onion>, <Peer:nsizc3h4pwq6xdkq.onion>, <Peer:mep3zrh575k5fw6g.onion>, <Peer:bhv7r6tgzmm2byqd.onion>, <Peer:rkj4wqo6o3czebvh.onion>, <Peer:m5eynz2x3onhxvtq.onion>, <Peer:5.9.9.18    >, <Peer:sttzj6imexwe7ydr.onion>, <Peer:wkidh6v3nc7ykgzb.onion>, <Peer:162.158.89.247>], 'workers_num': 0, 'time_added': 1523609851.606069, 'evt': <gevent.event.AsyncResult object at 0x7f408abd11d0>}
...
```

**Comment by HelloZeroNet** [comment]
Have you tried sign&publish data/users/content.json again?
Are the bigger files an optional files?

> If a merger site has the default 10MB size limit, and a user is trying to publish content that is, say 15MB. Is there any automatic fashion for the merger site's size limit to be increased while the user is publishing, or do we have to manually fileGet the merger's user content.json, modifying the max_size, signing and then sign/publish the content?

No, you won't be notified if an user want to publish larger files than the limit allows.

**Comment by anoadragon453** [comment]
> Have you tried sign&publish data/users/content.json again?

Yes, multiple times. One time I sign&published manually and noticed that `data.json` was *not* listed in `content.json`, even though it was in the folder at the time and shouldn't have been ignored. Signing/publishing through site resulted in `data.json` correctly being included. I'm not sure why exactly.

> Are the bigger files an optional files?

They weren't earlier (when I posted that second comment) but they are now. Curiously, when I connect a new device to the site (with 9MB optional files) and with `{data,content}.json` as required files, the client still downloads the 9MB but states that 0/9.8MB of optional files have been downloaded.

I think I'm just going to delete this merger site and try again, hopefully with less confusing results.

**Comment by anoadragon453** [comment]
So I've created a new merger site and am seeing the same issue. Here's what happens:

1. Remote Computer gets `data/users/xxx/content.json` which describes `data/users/xxx/data.json`.
2. Remote Computer requests `data/users/xxx/data.json`.
3. Remote Computer tries to patch `data/users/xxx/data.json` and cannot as the file does not exist:

```
[21:51:29] Site:14BBk5..nq1z Update for data/users/16HcyFK8526ZYJBKPpSDCbpJv6WWBNvENj/content.json looks valid, saving...                                                         [21:51:29] Site:14BBk5..nq1z Failed to patch data/users/16HcyFK8526ZYJBKPpSDCbpJv6WWBNvENj/data.json: [Errno 2] No such file or directory: u'data/14BBk5g3Fd3vTJjsfKS5pZGkzhCkdQnq1z/data/users/16HcyFK8526ZYJBKPpSDCbpJv6WWBNvENj/data.json'
```

So for some reason the daemon thinks the file is already there, but isn't and thus errors out. Should we instead be creating the file if it doesn't exist yet?

**Comment by HelloZeroNet** [comment]
What do you see in the log when you sign the `data/users/xxx/content.json` using the sidebar?
Does the modified entry changes?

**Comment by anoadragon453** [comment]
@HelloZeroNet Yes, it updates with a new hash, but what I'm seeing now is that it's updating with the **previous correct** size, which intuitively tells me that it may be signing `data.json` *then* changing the file. Let me check if this is the case.

**Comment by anoadragon453** [comment]
Still, it's odd that signing using the sidebar does not update the hash/size to the correct values. This is what the log shows when I do so:

```
[05:34:40] Site:14BBk5..nq1z Tracker error: http://retracker.spark-rostov.ru:80/announce (HTTP Error 403: Forbidden)
[05:34:40] Site:14BBk5..nq1z Recent peers 1 of 3 (need: 25)
[05:34:40] ConnServer Conn#30 5.9.9.18     [?] > Connecting...
[05:34:40] ConnServer Conn#31 3krzvwaaz2wusilh.onion [?] > Connecting...
[05:34:40] TorManager Creating new Tor socket to 3krzvwaaz2wusilh.onion:29593
[05:34:41] Site:14BBk5..nq1z Signing: data/users/content.json
[05:34:41] Site:14BBk5..nq1z Opening site data directory: data/14BBk5g3Fd3vTJjsfKS5pZGkzhCkdQnq1z/data/users/...
[05:34:41] Site:14BBk5..nq1z Changed files: [u'data/users/content.json']
[05:34:41] Site:14BBk5..nq1z Adding timestamp and sha512sums to new content.json...
[05:34:41] Site:14BBk5..nq1z Verifying private key...
[05:34:41] Site:14BBk5..nq1z Correct 14BBdiUy73vTJjsfKS5pZGkzhCkdQnq1z in valid signers: [u'14BBdiUy73vTJjsfKS5pZGkzhCkdQnq1z']
[05:34:41] Site:14BBk5..nq1z Signing data/users/content.json...
[05:34:41] Site:14BBk5..nq1z Saving to data/users/content.json...
[05:34:41] Site:yy..yy No database, waiting for dbschema.json...
[05:34:41] Db:yy Connected to data/yyy/merged-yy/yy.db in 0.000s (opened: 10, sqlite version: 2.6.0)...
[05:34:41] Db:yy Db check done in 0.001s, changed tables: []
[05:34:41] Site:1J3G8W..U2vx No database, waiting for dbschema.json...
[05:34:41] Db:yy Connected to data/yyy/merged-yy/yy.db in 0.000s (opened: 11, sqlite version: 2.6.0)...
[05:34:41] Db:yy Db check done in 0.000s, changed tables: []
[05:34:41] Site:14BBk5..nq1z File data/users/content.json signed!
[05:34:41] Site:14BBk5..nq1z Recent peers 1 of 3 (need: 10)
[05:34:41] Site:14BBk5..nq1z Publishing data/users/content.json to 5/4 peers (connected: 2) diffs: [] (0.00k)...
[05:34:41] ConnServer Conn#32 73kt3ioo4p6jgvbp.onion [?] > Connecting...
[05:34:41] TorManager Creating new Tor socket to 73kt3ioo4p6jgvbp.onion:28016
[05:34:41] ConnServer Conn#33 162.128.89.247 [?] > Connecting...
[05:34:41] ConnServer Conn#30 x.x.x.x     [?] > Crypt connection error: [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:661), adding ip x.x.x.x as broken ssl.
[05:34:41] ConnServer Conn#30 x.x.x.x     [v2] > Crypt out connection using: tls-rsa (server side: False, ping: 0.944s)...
```

That `broken ssl` error is also quite interesting.

`content.json` is now:

```
  6  "files": {
  7   "data.json": {
  8    "sha512": "db1e8ede98aef49ebf4dee682fa736651164f072d262c07e8f308c2f8a9b7d99",
  9    "size": 3914
 10   }
 11  },
```

And on the cmdline we see:

```
$ wc -c < 14BBdiUy73vTJjsfKS5pZGkzhCkdQnq1z/data/users/xxx/data.json 
4433
```

**Comment by HelloZeroNet** [comment]
based on the log you are signing the `data/users/content.json` file not the `data/users/xxx/content.json` one

**Comment by anoadragon453** [comment]
That's true, the sidebar is signing that, yeah. The site does sign the user's content.json though.

I've updated the code (there was a race condition) so that the size is now correct after the site signs it. The hash in `content.json` and sha512sum do match up as well now!

```
  6  "files": {
  7   "data.json": {
  8    "sha512": "fa86cd06379566bbc9410987659fec30547c169dbda8d6bd9f025e205a4e05ed",
  9    "size": 4953
 10   }
 11  }
```

```
$ sha512sum -b 14BBdiUy73vTJjsfKS5pZGkzhCkdQnq1z/data/users/xxx/data.json 
fa86cd06379566bbc9410987659fec30547c169dbda8d6bd9f025e205a4e05edafd0b0d06cd5b8530f8423970cc75072d81cf89e60deaa75d7b03d83995f8659 *14BBdiUy73vTJjsfKS5pZGkzhCkdQnq1z/data/users/xxx/data.json
$ wc -c < 14BBdiUy73vTJjsfKS5pZGkzhCkdQnq1z/data/users/xxx/data.json 
4953
```

And would ya know it, I now see the correct `data.json` on the server now. Phew!

So it was all due to a race condition in the code, where the site was signing/publishing before the `fileGet` and `fileWrite` commands for `data.json` completed. Ensuring that sign/publish was run only after these commands had completed eliminated the issue.

Thanks for the help, greatly appreciated!
